addons/print/models/print_printer.py

```diff
@@ -97,28 +97,45 @@ def _check_groups(self):
                                       printer.name)
 
     @api.multi
-    def printers(self, raise_if_not_found=False):
+    def printers(self, report_type=None, raise_if_not_found=False):
         """Determine printers to use"""
+        printers = self.browse()
 
-        # Start with explicitly specified list of printers, falling
-        # back to user's default printer, falling back to system
-        # default printer
-        printers = (self or self.env.user.printer_id or
-                    self.search([('is_default', '=', True),
-                                 ('group_id', '=', False)]))
+        if self and (report_type is None or self.report_type == report_type):
```

Comment:
Can self contain more than one element? In such case, `self.report_type` might fail.

Reply:
Yeah good spot, have cleaned up this method now. A filter is now used on `self` if it contains any records and `report_type` is specified:

```
printers = self
if report_type:
    printers = self.filtered(lambda p: p.report_type == report_type)
```